udata/core/pages/api.py

```diff
@@ -0,0 +1,46 @@
+from flask import request
+from flask_login import current_user
+
+from udata.api import API, api
+from udata.api_fields import patch
+
+from .models import Page
+
+ns = api.namespace("pages", "Pages related operations (beta)")
+
+common_doc = {"params": {"page": "The page ID or slug"}}
+
+
+@ns.route("/", endpoint="pages")
+class PagesAPI(API):
+    @api.secure
+    @api.doc("create_page", responses={400: "Validation error"})
+    @api.expect(Page.__write_fields__)
+    @api.marshal_with(Page.__read_fields__, code=201)
+    def post(self):
+        page = patch(Page(), request)
+        if not page.owner and not page.organization:
+            page.owner = current_user._get_current_object()
+
+        page.save()
+        return page, 201
+
+
+@ns.route("/<page:page>/", endpoint="page")
+class PageAPI(API):
+    @api.doc("get_page")
+    @api.marshal_with(Page.__read_fields__)
+    def get(self, page: Page):
+        return page
+
+    @api.secure
+    @api.doc("update_page")
+    @api.expect(Page.__write_fields__)
+    @api.marshal_with(Page.__read_fields__)
+    def put(self, page: Page):
+        page = patch(Page(), request)
+        if not page.owner and not page.organization:
+            page.owner = current_user._get_current_object()
```

Comment:
I'm confused about this logic on a put?

Reply:
Yes it may not be necessary, copied from the `POST`. But I'm not sure why it's necessary in the `POST` too. I think it could be handle but the `Owned` class since we always want to do that in all classes using these attributes…

Reply:
I thought the same on the `POST` too :p But I saw you did this on the reuse POST API too and didn't check if there was a reason for it or not :p

Reply:
It is necessary for the `POST` endpoint because our validation allow organization and owner to be empty at the same time.
For this one, it is not necessary because this is not really a PUT, we are patching the existing page so there is always an owner or an organization.